tests/acceptance/standard/test_hooks_processor.py

```diff
@@ -0,0 +1,63 @@
+import logging
+
+from tests.acceptance import run_gitlabform, create_group, create_project
+
+
+class TestHooksProcessor:
+    def test_project_hooks_create_delete(self, gl, project, project_hook):
+        target = project.path_with_namespace
+        test_yaml = f"""
+            projects_and_groups:
+              {target}:
+                hooks:
+                  {project_hook.url}:
+                    push_events: {project_hook.push_events}
+                    merge_requests_events: {project_hook.merge_requests_events}
+            """
+
+        run_gitlabform(test_yaml, target)
+
+        created_hook = project.hooks.get(project_hook.id)
+        assert project_hook == created_hook
+
+        delete_yaml = f"""
+        projects_and_groups:
+            {target}:
+              hooks:
+                {created_hook.url}:
+                  delete: true
+        """
+
+        run_gitlabform(delete_yaml, target)
+        hooks = project.hooks.list()
+
+        assert len(hooks) == 0
+
+    def test_project_hooks_update(gl, project, project_hook):
+        target = project.path_with_namespace
+        test_yaml = f"""
+            projects_and_groups:
+              {target}:
+                hooks:
+                  {project_hook.url}:
+                    push_events: {project_hook.push_events}
+                    merge_requests_events: {project_hook.merge_requests_events}
+            """
+
+        original_hook = project.hooks.create(
+            {
+                "url": project_hook.url,
+                "push_events": not project_hook.push_events,
+                "merge_requests_events": not project_hook.merge_requests_events,
+            }
+        )
+        project_hook.id = original_hook.id
+
+        run_gitlabform(test_yaml, target, include_archived_projects=False)
+
+        modified_hook = project.hooks.get(original_hook.id)
+
+        assert modified_hook.asdict() != original_hook.asdict()
+        assert modified_hook == project_hook
+
+        modified_hook.delete()
```

Comment:
Not sure if it's just me, but this test seems a little unintuitive. What do you think about calling `run_gitlabform` twice with different config to the hook.

After the first call to `run_gitlabform`, retrieve the hook and then call `run_gitlabform` again with a modified yaml config. Then retrieve the hook again. Compare the before and after result.

Reply:
Ah, I see! When separating the create and the update, because I didn't want to muddy the waters I had created the hook to modify without passing through gitlabform... I thought let's test this and only this part of run() -> hooks_processor. But I can see how this might be a bit convoluted (I can convince myself of any rationale, that's the problem).

Reply:
I see. Thanks for discussing it. Changes are looking good so far.